src/game.py

```diff
@@ -114,16 +151,33 @@ def handle_winning_game(self):
     def handle_draw(self):
         self.ui.display_message(self.message.game_over_message())
 
-    def take_turns(self):
+    def take_turns_human_vs_human(self):
         self.prompt_for_move(self.total_marks_on_board)
         self.handle_mark_board()
         self.total_marks_on_board = self.board.count_marks(
             self.game_board, self.player_one, self.player_two
         )
         self.get_formatted_board()
 
+    def take_turns_comp_vs_human(self):
+        self.prompt_for_move(self.total_marks_on_board)
+        self.handle_mark_board()
+        self.ui.display_message(self.message.computer_took_turn())
+        self.handle_computer_mark_board()
+        self.total_marks_on_board = self.board.count_marks(
+            self.game_board, self.player_one, self.player_two
+        )
+        self.get_formatted_board()
+
+    def take_turns(self):
+        if self.play_against_computer:
+            self.take_turns_comp_vs_human()
+        else:
+            self.take_turns_human_vs_human()
```

Comment:
Do you think it would be possible to use duck typing to implement the same interface with the types of players, so you don't need to distinguish between a computer player and a human player in the logic? Does the `Game` class need to be able to distinguish between a computer and a human to manage the game?

Reply:
Would this look like:
Creating two classes, `ComputerPlayer` and `HumanPlayer`, and then both of them have a method, `take_turns`?  Then you just call `take_turns`?  

![Screen Shot 2022-05-05 at 9 54 21 AM](https://user-images.githubusercontent.com/73620216/166951406-22263129-eba1-4793-85b6-067d72a237ee.png)

Something like above?  (This doesn't actually work but just trying to see if I'm even on the right track or far off).  Maybe not since there is still the logic distinguishing between the players so now I'm not so sure.

Then I get hung up on a few things - I tried extracting`take_turns` methods and putting them into their respective classes, but they use several other methods from `Game` in those methods.

Also, I'm having a hard time visualizing how I would not distinguish between players in the logic...

Your question leads me to believe that maybe `Game` would not need to be able to distinguish between human and computer but I'm a little stuck on this.

Reply:
a `Game` should just be able to keep track of `player1` and `player2` and treat them exactly the same way all the time.  That's [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) at work.

`player1` could be a "human" or a "computer"
`player2` could be a "human" or a "computer"

the `Game` should not know, or care about the internals of the player, and only keep track of the objects themself.

Reply:
with this kind of abstraction, `take_turns()` would not have to change.  It would just be going back and forth between `player1` and `player2` like it was already doing.